### PR TITLE
Documentation: add more details to Interferometer initialization functions

### PR DESCRIPTION
**Pull request by AroosaIjaz**
**Description of the Change:**

I added a little more detail to the description of Interferometer parameter initialization functions `interferometer_normal` and `interferometer_uniform`. 

**Benefits:**

Avoids any confusion and helps the user decipher the output of these functions without having to look up Interferometer function itself.

### CODE REVIEW

File: pennylane/init.py [hunk 1]
```diff
@@ -495,7 +495,7 @@ def cvqnn_layer_normal(n_wires, mean=0, std=1, mean_active=0, std_active=0.1, se
 def interferometer_uniform(n_wires, low=0, high=2 * pi, seed=None):
     r"""Creates a list of three parameter arrays for :func:`~.Interferometer`, drawn from a uniform distribution.
 
-    The shape of the arrays is either ``(n_wires,)`` or ``(n_wires*(n_wires-1)/2,)``.
+    The shape of one of the arrays is ``(n_wires,)``. It contains the initialized rotation angles :math:`\varphi` for the n_wires Rotation gates. The shape of other two arrays is ``(n_wires*(n_wires-1)/2,)``. These contain the initialized transmittivity angles :math:`\theta` and phase angles :math:`\phi` for the Beam Splitter gates.
```
Comment (co9olguy) [inline pennylane/init.py]:
```suggestion
    The shape of one of the arrays is ``(n_wires,)``. It contains the initialized rotation angles :math:`\varphi` for the ``n_wires`` Rotation gates. The other two arrays have shapes ``(n_wires*(n_wires-1)/2,)``. These arrays contain the initialized transmittivity angles :math:`\theta` and phase angles :math:`\phi` for the Beam Splitter gates. 
```
Note: it is rendering poorly on my github browser page, but there are 3 changes suggested in this paragraph

Comment (co9olguy) [reply pennylane/init.py]: If the arrays come out in a particular order (e.g., the first one is the one with shape `(n_wires,)`), we might as well say that.

Comment (co9olguy) [reply pennylane/init.py]: iirc, the rotation gates appear at the end, don't they? 
In that case (please verify), we can say something like "the final ``n_wires`` Rotation gates".

File: pennylane/init.py [hunk 2]
```diff
@@ -524,7 +524,7 @@ def interferometer_uniform(n_wires, low=0, high=2 * pi, seed=None):
 def interferometer_normal(n_wires, mean=0, std=0.1, seed=None):
     r"""Creates a list of three parameter arrays for :func:`~.Interferometer`, drawn from a normal distribution.
 
-    The shape of the arrays is either ``(n_wires,)`` or ``(n_wires*(n_wires-1)/2,)``.
+    The shape of one of the arrays is ``(n_wires,)``. It contains the initialized rotation angles :math:`\varphi` for the n_wires Rotation gates. The shape of other two arrays is ``(n_wires*(n_wires-1)/2,)``. These contain the initialized transmittivity angles :math:`\theta` and phase angles :math:`\phi` for the Beam Splitter gates.
```
Comment (co9olguy) [inline pennylane/init.py]: same comments/changes/suggestions as above

File: pennylane/init.py (hunk 1 above)
Comment (josh146) [reply pennylane/init.py]: Could we be even more explicit? Something along the lines of:

```rest
This function returns a list of the form ``[phi, theta, varphi]`` where:

* ``phi`` is the list of beamsplitter angles, of size ``(n_wires*(n_wires-1)/2, )``

* ``theta`` is the list of beamsplitter phases, of size ``(n_wires*(n_wires-1)/2, )``

* ``varphi`` is the list of local phases, of size ``(n_wires, )``
```

Making it explicit and putting it in a bulletpoint list makes it much more readable to me.

Comment (AroosaIjaz) [reply pennylane/init.py]: yes the rotation gate parameters come at the end.

Comment (AroosaIjaz) [reply pennylane/init.py]: > It contains the initialized rotation angles :math:`\varphi` for the  ``n_wires`` Rotation gates

I checked this with the rest of the documentation when writing:  Rendering text within accented apostrophes  `` ... ``  is reserved for function names, arguments to functions and code snippets used outside a  coding environment.  So I did not use this font for n_wires as it is being refereed to as a number here. Should I still go ahead with this?

Comment (AroosaIjaz) [reply pennylane/init.py]: > Could we be even more explicit? Something along the lines of:
> 
> ```
> This function returns a list of the form ``[phi, theta, varphi]`` where:
> 
> * ``phi`` is the list of beamsplitter angles, of size ``(n_wires*(n_wires-1)/2, )``
> 
> * ``theta`` is the list of beamsplitter phases, of size ``(n_wires*(n_wires-1)/2, )``
> 
> * ``varphi`` is the list of local phases for the final rotation gates, of size ``(n_wires, )``
> ```
> 
> Making it explicit and putting it in a bulletpoint list makes it much more readable to me.

Yeah, this is easier to read.

Comment (co9olguy) [reply pennylane/init.py]: > > It contains the initialized rotation angles :math:`\varphi` for the  `n_wires` Rotation gates
> 
> I checked this with the rest of the documentation when writing: Rendering text within accented apostrophes `...` is reserved for function names, arguments to functions and code snippets used outside a coding environment. So I did not use this font for n_wires as it is being refereed to as a number here. Should I still go ahead with this?

In this case, it's a code snippet (i.e., the argument to the function) appearing outside a coding environment.

File: pennylane/init.py [hunk 3]
```diff
@@ -493,11 +493,15 @@ def cvqnn_layer_normal(n_wires, mean=0, std=1, mean_active=0, std_active=0.1, se
...
 
 def interferometer_uniform(n_wires, low=0, high=2 * pi, seed=None):
-    r"""Creates a list of three parameter arrays for :func:`~.Interferometer`, drawn from a uniform distribution.
+    r"""Returns a list of three arrays of the form ``[theta, phi, varphi]`` where:
 
-    The shape of the arrays is either ``(n_wires,)`` or ``(n_wires*(n_wires-1)/2,)``.
+    * ``theta`` is the list of beam splitter transmittivity angles, of size ``(n_wires*(n_wires-1)/2, )``
 
-    The parameters are initialized uniformly from the interval ``[low, high]``.
+    * ``phi`` is the list of beam splitter phases, of size ``(n_wires*(n_wires-1)/2, )``
```
Comment (co9olguy) [inline pennylane/init.py]:
```suggestion
    * ``phi`` is the list of beam splitter phases, of size ``(n_wires*(n_wires-1)/2, )``
```

Comment (co9olguy) [inline pennylane/init.py]: here it says "list", but below it says "array" (same for other angles). Which is it?

File: pennylane/init.py [hunk 4]
```diff
@@ -493,11 +493,15 @@ def cvqnn_layer_normal(n_wires, mean=0, std=1, mean_active=0, std_active=0.1, se
 
 
 def interferometer_uniform(n_wires, low=0, high=2 * pi, seed=None):
-    r"""Creates a list of three parameter arrays for :func:`~.Interferometer`, drawn from a uniform distribution.
+    r"""Returns a list of three parameter arrays of the form ``[theta, phi, varphi]``, where:
 
-    The shape of the arrays is either ``(n_wires,)`` or ``(n_wires*(n_wires-1)/2,)``.
+    * ``theta`` is the list of beam splitter transmittivity angles, of size ``(n_wires*(n_wires-1)/2, )``
```
Comment (co9olguy) [inline pennylane/init.py]:
```suggestion
    * ``theta`` is the list of beamsplitter transmittivity angles, of size ``(n_wires*(n_wires-1)/2, )``
```

File: pennylane/init.py [hunk 5]
```diff
@@ -493,11 +493,15 @@ def cvqnn_layer_normal(n_wires, mean=0, std=1, mean_active=0, std_active=0.1, se
...
 
-    The parameters are initialized uniformly from the interval ``[low, high]``.
+    * ``phi`` is the list of beam splitter phases, of size ``(n_wires*(n_wires-1)/2, )``
```
Comment (co9olguy) [inline pennylane/init.py]:
```suggestion
    * ``phi`` is the list of beamsplitter phases, of size ``(n_wires*(n_wires-1)/2, )``
```

File: pennylane/init.py [hunk 6]
```diff
@@ -522,10 +526,14 @@ def interferometer_uniform(n_wires, low=0, high=2 * pi, seed=None):
 
 
 def interferometer_normal(n_wires, mean=0, std=0.1, seed=None):
-    r"""Creates a list of three parameter arrays for :func:`~.Interferometer`, drawn from a normal distribution.
+    r"""Returns a list of three parameter arrays of the form ``[theta, phi, varphi]``, where:
+
+    * ``theta`` is the array of beam splitter transmittivity angles, of size ``(n_wires*(n_wires-1)/2, )``
```
Comment (co9olguy) [inline pennylane/init.py]:
```suggestion
    * ``theta`` is the array of beamsplitter transmittivity angles, of size ``(n_wires*(n_wires-1)/2, )``
```

File: pennylane/init.py [hunk 7]
```diff
@@ -522,10 +526,14 @@ def interferometer_uniform(n_wires, low=0, high=2 * pi, seed=None):
...
 
-    The shape of the arrays is either ``(n_wires,)`` or ``(n_wires*(n_wires-1)/2,)``.
+    * ``phi`` is the array of beam splitter phases, of size ``(n_wires*(n_wires-1)/2, )``
```
Comment (co9olguy) [inline pennylane/init.py]:
```suggestion
    * ``phi`` is the array of beamsplitter phases, of size ``(n_wires*(n_wires-1)/2, )``
```

File: pennylane/init.py (hunk 3 above)
Comment (AroosaIjaz) [reply pennylane/init.py]: The function returns a list of 3 arrays. it should be `array` here. will fix it

**Comment by AroosaIjaz** [comment]
This is ready for a final review and merge.

**Comment by co9olguy** [comment]
Not sure what the deal is with codefactor. Merging this now

**Pull request merged**